src/test/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProviderTest.java

```diff
@@ -0,0 +1,279 @@
+import com.epam.grid.engine.cmd.GridEngineCommandCompiler;
+import com.epam.grid.engine.cmd.SimpleCmdExecutor;
+import com.epam.grid.engine.entity.CommandResult;
+import com.epam.grid.engine.entity.CommandType;
+import com.epam.grid.engine.entity.QueueFilter;
+import com.epam.grid.engine.entity.queue.Queue;
+import com.epam.grid.engine.entity.queue.QueueVO;
+import com.epam.grid.engine.entity.queue.SlotsDescription;
+import com.epam.grid.engine.exception.GridEngineException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.doReturn;
+
+@SpringBootTest
+@TestPropertiesWithSlurmEngine
+public class SlurmQueueProviderTest {
+
+    private static final List<String> validStdout = List.of("newPartition|worker1|1|ALL");
+    private static final List<String> validListOut = List.of("normal|worker1|1|ALL", "partition1||worker[21-2]|admin",
+            "partition2|worker2|1|slurm");
+    private static final List<String> validStdoutUpdatedPartition = List.of("updatingPartitionName|worker1,worker2|"
+            + "2|ALL");
+    private static final String[] CREATE_COMMAND_PATTERN = new String[]{"scontrol CREATE PartitionName=newPartition"
+            + "AllowGroups=ALL Nodes=worker1"};
+    private static final String[] LIST_COMMAND_PATTERN = new String[]{"sinfo -h -o \"%R|%N|%c|%g\""};
+    private static final String[] DELETE_COMMAND_PATTERN = new String[]{"scontrol DELETE "
+            + "PartitionName=deletedPartiiton"};
+    private static final String[] UPDATE_COMMAND_PATTERN = new String[]{"scontrol UPDATE "
+            + "PartitionName=updatingPartitionName AllowGroups=ALL Nodes=worker1,worker2"};
+    private static final String[] SINFO_COMMAND_PATTERN = new String[]{"sinfo --partition=updatingPartitionName "
+            + " -h -o \"%R|%N|%c|%g\""};
+    private static final String SCONTROL_COMMAND_TEMPLATE = "scontrolCommand";
+    private static final String SINFO_COMMAND_TEMPLATE = "sinfo";
+    private static final List<String> ownerList = List.of("root", "slurm");
+    private static final List<String> parallelEnvList = List.of("someVar1", "someVar1");
+    private static final List<String> nodeListOneNode = List.of("worker1");
+    private static final List<String> nodeListTwoNodes = new ArrayList<>(List.of("worker1", "worker2"));
+    private static final List<String> groupsList = new ArrayList<>(List.of("ALL"));
```

Comment:
`new ArrayList<>()` is redundant here

Reply:
if lists with nodes and groups will be immutable, we will get an exception on sorting during partition update

Reply:
probably we could change the logic of the provider method
from (127 line) `final List<String> updateUserGroups = updateRequest.getAllowedUserGroups()` to
```
updateRequest.getAllowedUserGroups().stream()
     .sorted()
     .collect(Collectors.toList());
```
the same with nodes

Reply:
done